yjit/src/codegen.rs

```diff
@@ -2342,6 +2342,11 @@ fn gen_setinstancevariable(
                 if needs_extension {
                     // Generate the C call so that runtime code will increase
                     // the capacity and set the buffer.
+                    asm.comment("call rb_ensure_iv_list_size");
+
+                    // It allocates so can trigger GC, which takes the VM lock
+                    // so could yield to a differnt ractor.
+                    jit_prepare_routine_call(jit, asm);
                     asm.spill_temps(); // for ccall
```

Comment:
This `asm.spill_temps()` is dominated by the other call above on line 2302. Maybe one of the calls should be removed?

Reply:
That seems right. 

`jit_prepare_routine_call` calls `asm.spill_temps()` internally, so your change should obviate it anyway.

```suggestion
```